server/src/routes/schedule.js

```diff
@@ -57,15 +59,35 @@ router.get('/candidates', async (req, res) => {
   });
 });
 
+/**
+ * Get a candidate's firstname, for the candidate availabilty page.
+ *
+ * This should be the only method that can be used without authentication.
+ */
+router.get('/candidate/name/:uuid', (req, res) => {
+  const { uuid } = req.params;
+  const sql = 'SELECT firstName FROM Candidate WHERE uuid = ?';
+  const sqlcmd = connection.format(sql, [uuid]);
+  connection.query(sqlcmd, (err, result) => {
+    if (err) {
+      return res.status(400).send({ message: 'Internal server Error.' });
+    }
+    if (result.length === 0) {
+      return res.status(204).send({ message: 'No candidate' });
+    }
+    // Just send the UUID and the first name
+    return res.send([{ uuid, firstName: result[0].firstName }]);
+  });
+});
 
 // get a specific candidate
-router.get('/candidate/:id', (req, res) => {
-  const { id } = req.params;
-  const sql = 'SELECT * FROM Candidate WHERE candidateId = ?';
-  const sqlcmd = connection.format(sql, [id]);
+router.get('/candidate/:uuid', notAuthMiddleware, (req, res) => {
+  const { uuid } = req.params;
+  const sql = 'SELECT * FROM Candidate WHERE uuid = ?';
+  const sqlcmd = connection.format(sql, [uuid]);
   connection.query(sqlcmd, (err, result) => {
     if (err) {
-      throw err;
+      return res.status(400).send({ message: 'Internal server error.' });
```

Comment:
I think the status code for internal server error should be 5xx.

Reply:
Changed to 500